### ISSUE TITLE
Schema.clone() - TypeError: Cannot use 'in' operator to search for 'defaults' in undefined

### ISSUE THREAD

**Issue by SteffenLanger** [Automattic/mongoose#11455]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
* Create a Mongoose schema with sub schemas.
* Clone the top-level schema and create a model from it.
* Use Model.updateMany() to update fields within the subschemas.
* 🐞 An error is thrown: `TypeError: Cannot use 'in' operator to search for 'defaults' in undefined`

This error is not thrown if the schema for the model is not cloned.

[This might be a related issue](https://github.com/Automattic/mongoose/issues/11394) but Mongoose does not run into the TypeError if the original schema is used (vs. the cloned schema).

**If the current behavior is a bug, please provide the steps to reproduce.**

I created a simple GitHub repo: https://github.com/SteffenLanger/mongoose-bug-clone-schema

This is the content of the index.js file:
```javascript
const mongoose = require('mongoose');

mongoose.connect('mongodb://localhost:27017/myapp');

const bankAccountSchema   = new mongoose.Schema({
    iban : String,
    bic  : String
});
const contactPersonSchema = new mongoose.Schema({
    firstName   : String,
    lastName    : String,
    bankAccount : bankAccountSchema
});
const companySchema       = new mongoose.Schema({
    _id              : String,
    name             : String,
    contactPerson    : contactPersonSchema,
    _documentVersion : Number
}, {
    timestamps : true
});

/**
 * This is the issue: If companySchema is used instead of companySchema.clone(), updateMany works fine.
 */
const CompanyModel = mongoose.model('Company', companySchema.clone());

(async function () {
    await CompanyModel.create({
        _id              : 'abcde',
        name             : 'ACME',
        _documentVersion : 1,
    });
    await CompanyModel.updateMany({
            "_documentVersion" : 1,
            "$and"             : [
                {
                    "_id" : "abcde"
                }
            ]
        }, {
            name          : 'Other Company name',
            contactPerson : {
                firstName   : 'John',
                lastName    : 'Doe',
                bankAccount : {
                    iban : 'abc',
                    bic  : 'def'
                }
            }
        }, {
            "multi"         : false,
            "runValidators" : true,
            "context"       : "query"
        })
        .lean(true)
        .exec();
})();
```

**What is the expected behavior?**
The cloned schema should behave exactly like the original schema.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node.js: v16.13.1
Mongoose: 6.2.3
MongoDB: 5.0.5

**Comment by IslandRhythms** [comment]
```javascript
const mongoose = require('mongoose');



const bankAccountSchema   = new mongoose.Schema({
    iban : String,
    bic  : String
});
const contactPersonSchema = new mongoose.Schema({
    firstName   : String,
    lastName    : String,
    bankAccount : bankAccountSchema
});
const companySchema       = new mongoose.Schema({
    _id              : String,
    name             : String,
    contactPerson    : contactPersonSchema,
    _documentVersion : Number
}, {
    timestamps : true
});

/**
 * This is the issue: If companySchema is used instead of companySchema.clone(), updateMany works fine.
 */
const CompanyModel = mongoose.model('Company', companySchema.clone());

(async function () {

    await mongoose.connect('mongodb://localhost:27017/myapp');
    await mongoose.connection.dropDatabase();
    await CompanyModel.create({
        _id              : 'abcde',
        name             : 'ACME',
        _documentVersion : 1,
    });
    await CompanyModel.updateMany({
            "_documentVersion" : 1,
            "$and"             : [
                {
                    "_id" : "abcde"
                }
            ]
        }, {
            name          : 'Other Company name',
            contactPerson : {
                firstName   : 'John',
                lastName    : 'Doe',
                bankAccount : {
                    iban : 'abc',
                    bic  : 'def'
                }
            }
        }, {
            "multi"         : false,
            "runValidators" : true,
            "context"       : "query"
        })
        .lean(true)
        .exec();
        console.log(await CompanyModel.find())
})();
```

**Comment by vkarpov15** [comment]
I took a look and confirmed that this is the same issue that we fixed in #11394. Fix will be in v6.2.4 :+1:
